core/shared_queue.rs

```diff
@@ -152,17 +153,19 @@ impl SharedQueue {
     Some(&self.bytes[off..end])
   }
 
-  pub fn push(&mut self, record: &[u8]) -> bool {
+  pub fn push(&mut self, promise_id: c_int, record: &[u8]) -> bool {
```

Comment:
What happens if there is no room in the shared_queue ?

Reply:
It gets passed here
https://github.com/afinch7/deno/blob/608d6c571af166ba52db0ad5b82c4d608ea7e2b4/core/isolate.rs#L618-L623